### PR TITLE
Move application into app directory rather then builds

### PR DESCRIPTION
**Pull request by rhatdan**
Builds is too generic and difficult to understand. Also move all app content into the build directory to make it easier to find and work with.

### CODE REVIEW

File: .github/workflows/ailab_image_build_push.yaml [hunk 1]
```diff
@@ -286,8 +286,8 @@ jobs:
           image: ${{ env.SUMMARIZER_IMAGE_NAME }}
           tags: latest ${{ github.sha }}
           platforms: linux/amd64, linux/arm64
-          containerfiles: ./recipes/natural_language_processing/summarizer/builds/Containerfile
-          context: recipes/natural_language_processing/summarizer
+          containerfiles: ./recipes/natural_language_processing/summarizer/app/Containerfile
+          context: recipes/natural_language_processing/summarizer/app/app
```
Comment (sallyom) [inline .github/workflows/ailab_image_build_push.yaml]: extra `app` here

File: .github/workflows/build-images.yaml [hunk 2]
```diff
@@ -83,9 +83,9 @@ jobs:
           image: ${{ env.CODEGEN_IMAGE }}
           tags: latest ${{ github.sha }}
           platforms: linux/amd64, linux/arm64
-          context: recipes/natural_language_processing/code-generation
+          context: recipes/natural_language_processing/code-generation/app
```
Comment (sallyom) [inline .github/workflows/build-images.yaml]: the folder is `natural_language_processing/codegen` not code-generation

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -64,14 +64,14 @@ application:
...
-### builds/Containerfile
+### app/Containerfile
 
 This will be the Containerfile used to build the client side image of your AI application. Whenever possible, we will use Red Hat's UBI as our base image. Below please see an example from the chatbot recipe.   
 
 ```Dockerfile
 FROM registry.access.redhat.com/ubi9/python-311:1-52
 WORKDIR /chat
-COPY builds/requirements.txt .
+COPY requirements.txt .
```
Comment (sallyom) [inline CONTRIBUTING.md:74]: app/requirements.txt, right?

Comment (sallyom) [reply CONTRIBUTING.md:74]: nm, I see this file is in app now

File: .github/workflows/ailab_image_build_push.yaml (hunk 1 above)
Comment (rhatdan) [reply .github/workflows/ailab_image_build_push.yaml]: Fixed

File: .github/workflows/build-images.yaml (hunk 2 above)
Comment (rhatdan) [reply .github/workflows/build-images.yaml]: Fixed. I just found that myself.

File: CONTRIBUTING.md (hunk 3 above)
Comment (rhatdan) [reply CONTRIBUTING.md:74]: No if we move the contextdir to the `app` subdir then you just refer to the requirements.txt

**Review by sallyom — APPROVED** [review]
LGTM!

**Pull request merged**